ray-operator/controllers/ray/utils/util.go

```diff
@@ -508,3 +509,27 @@ func FindContainerPort(container *corev1.Container, portName string, defaultPort
 	}
 	return defaultPort
 }
+
+// IsJobFinished checks whether the given Job has finished execution.
+// It does not discriminate between successful and failed terminations.
+// src: https://github.com/kubernetes/kubernetes/blob/a8a1abc25cad87333840cd7d54be2efaf31a3177/pkg/controller/job/utils.go#L26
+func IsJobFinished(j *batchv1.Job) bool {
+	for _, c := range j.Status.Conditions {
+		if (c.Type == batchv1.JobComplete || c.Type == batchv1.JobFailed) && c.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func EnvInt64(name string, fallback int64) int64 {
```

Comment:
Perhaps we could directly use `strconv.ParseInt(v, 10, 64)` in the `buildRedisCleanupJob` function. If users input invalid values, it should fail. In open-source projects, we often need to address numerous user questions without access to their Kubernetes clusters. Therefore, identifying and surfacing issues as early as possible might be a better strategy for maintainers.

Reply:
Hi @kevin85421,

I find it hard to make the operator fail in the `buildRedisCleanupJob`. This function is called only when the operator is going to create the cleanup job, which is already too late.

Considering that the `GCS_FT_REDIS_CLEANUP_DEADLINE_SECONDS` is an environment variable, the proper timing to fail the operator should be during initialization. How about we parse the env and set the parsed result as a member field of RayClusterReconciler?

Reply:
After our latest discussion, we decided not to introduce the `GCS_FT_REDIS_CLEANUP_DEADLINE_SECONDS` but always set the `.spec.activeDeadlineSeconds` to 300 seconds instead.